src/EventSubscriber/RequestSubscriber.php

```diff
@@ -271,12 +283,63 @@ protected function constructQuerySpan($connection, $driver, array $query) {
    *   An array of options to pass to the PHP Agent. Ie. tags.
    */
   protected function prepareAgentOptions() {
+    $tags = [];
+
     $route_options = $this->routeMatch->getRouteObject()->getOptions();
     if (isset($route_options['_admin_route'])) {
-      return ['tags' => ['is_admin_route' => TRUE]];
+      $tags['is_admin_route'] = TRUE;
+    }
+
+    // Add tags based on the path patterns configured.
+    $tags = $tags + $this->preparePathPatternTags();
+
+    return $tags;
+  }
+
+  /**
+   * Provide tags based on the path patterns configured.
+   *
+   * @return array
+   *   An array of tags to pass to the PHP Agent.
+   */
+  protected function preparePathPatternTags() {
+    $tags = [];
+
+    // Fetch the current path from route object.
+    $route_object = $this->routeMatch->getRouteObject();
+    $path = $route_object->getPath();
+
+    // Fetch the configured path patterns.
+    $tag_config = $this->apiService->getTagConfig();
+    $path_pattern = $tag_config['path_patterns'];
+
+    if (empty($path_pattern)) {
+      return;
+    }
+
+    // Explode path patterns by new line.
+    $path_patterns = explode(PHP_EOL, $path_pattern);
+
+    // Add tags depending on the path pattern set.
+    foreach ($path_patterns as $path_pattern) {
```

Comment:
**CHANGE REQUEST**

It's not good practice in terms of writing clear code and code that is not but-prone to reuse names for unrelated variables. `$path_pattern` defined above is different than the `$path_pattern` within the loop, name it differently: `foreach ($path_patterns as $pattern) {`.